### PR TITLE
feat: use bitcoin-core as the block notifier

### PR DESCRIPTION
**Pull request by djordon**
## Description

Closes: https://github.com/stacks-network/sbtc/issues/383.

The electrum client that we have for rust requires you to poll the server for bitcoin blocks. While not terrible, using the ZeroMQ interface on bitcoin-core gives us real-time notifications, and allows us to remove electrum server as a requirement for running the signer application.

## Changes

* Add structs subscribe to bitcoin-cores ZeroMQ notification interface for block and block hash notifications.
* Add an integration test that listens to bitcoin-core messages and checks the new code.

## Testing Information

The two integration tests check the new logic that was added here. It does not validate the messages received and does not reproduce the logic laid out in the bitcoin docs for what happens during a reorg. That we should do later in https://github.com/stacks-network/sbtc/issues/405.

## Checklist:

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: signer/tests/integration/zmq.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            if sx.is_closed() {
+                break;
+            }
+
+            tracing::info!("Sending block {:?}", block.block_hash());
+            sx.send(block).await.unwrap();
+        }
+    });
+
+    // When the faucet generates a blocks it returns the block hash of the
```
Comment (xoloki) [inline signer/tests/integration/zmq.rs:43]: generates a block

File: signer/tests/integration/zmq.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            if sx.is_closed() {
+                break;
+            }
+
+            tracing::info!("Sending block hash {block_hash}");
+            sx.send(block_hash).await.unwrap();
+        }
+    });
+
+    // When the faucet generates a blocks it returns the block hash of the
```
Comment (xoloki) [inline signer/tests/integration/zmq.rs:97]: generates a block

**Review by xoloki — APPROVED** [review]
Lgtm, just a few comment corrections. I really like the zmq interface.

File: signer/tests/integration/zmq.rs (hunk 1 above)
Comment (djordon) [reply signer/tests/integration/zmq.rs]: https://github.com/stacks-network/sbtc/pull/406/commits/4cebe35e4f5d8d3edd90818224ba280402d4c374

File: signer/tests/integration/zmq.rs (hunk 2 above)
Comment (djordon) [reply signer/tests/integration/zmq.rs]: https://github.com/stacks-network/sbtc/pull/406/commits/4cebe35e4f5d8d3edd90818224ba280402d4c374

**Pull request merged**